### PR TITLE
CR102 Validation Refactor

### PR DESCRIPTION
**Pull request by noydavidi**
## Description
Moved CR102 to the new validate format.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/CR_validators/CR102_validate_execution_mode_search_window.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ValidationResult,
+)
+
+ContentTypes = CorrelationRule
+
+
+class ExecutionModeSearchWindowValidator(BaseValidator[ContentTypes]):
+    error_code = "CR102"
+    description = "Validates 'search_window' existence and non-emptiness for 'execution_mode' = 'SCHEDULED'."
+    rationale = ""
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/CR_validators/CR102_validate_execution_mode_search_window.py:17]: ?

File: demisto_sdk/commands/validate/validators/CR_validators/CR102_validate_execution_mode_search_window.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            if(content_item.execution_mode == "REAL_TIME"): #when execution_mode is set to REAL_TIME, search_window can be empty
+                continue
+            elif((not content_item.search_window) or (content_item.execution_mode == "SCHEDULED" and not content_item.search_window)):
+                validation_results.append(ValidationResult(
+                validator=self,
+                message=self.error_message,
+                content_object=content_item,
+            ))
+                
+        return validation_results
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/CR_validators/CR102_validate_execution_mode_search_window.py:34]: Let's do this in list comprehension.

File: demisto_sdk/commands/content_graph/objects/correlation_rule.py [hunk 3]
```diff
@@ -7,6 +10,9 @@
 
 
 class CorrelationRule(ContentItemXSIAM, content_type=ContentType.CORRELATION_RULE):  # type: ignore[call-arg]
+    execution_mode: Optional[str] = Field(None)
```
Comment (YuvHayun) [inline demisto_sdk/commands/content_graph/objects/correlation_rule.py:13]: Let's make this field an enum with possible values 'REAL_TIME', 'SCHEDULED'.
You can see PlaybookTaskType in base_playbook object as a reference.
Make sure to fix the validation and test accordingly.

File: demisto_sdk/commands/validate/tests/CR_validators_test.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+def test_validate_execution_mode_search_window_with_no_fields():
+    """
+    Given:
+        An default correlation rule with no search_window and execution_mode fields.
+    When:
+        Calling Validate
+    Then:
+        The validation should fail
+    """
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/CR_validators_test.py:18]:
```suggestion
    When:
        Calling Validate.
    Then:
        The validation should fail.
    """
```

File: demisto_sdk/commands/validate/tests/CR_validators_test.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    
+
+def test_validate_execution_mode_search_window_with_empty_search_window():
+    """
+    Given:
+        A correlation rule with empty search_window but execution_mode == SCHEDULED.
+    When:
+        Calling Validate
+    Then:
+        The validation should fail
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/CR_validators_test.py:30]:
```suggestion
        Calling Validate.
    Then:
        The validation should fail.
```

File: demisto_sdk/commands/validate/tests/CR_validators_test.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def test_validate_execution_mode_search_window_with_null_search_window():
+    """
+    Given:
+        A correlation rule with empty search_window but execution_mode == SCHEDULED.
+    When:
+        Calling Validate
+    Then:
+        The validation should fail
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/CR_validators_test.py:47]:
```suggestion
        Calling Validate.
    Then:
        The validation should fail.
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Great job!
See my comments.

**Review by YuvHayun — APPROVED** [review]
Approved from my side.

File: demisto_sdk/commands/validate/validators/CR_validators/CR102_validate_execution_mode_search_window.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    is_auto_fixable = False
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self, message=self.error_message, content_object=content_item
+            )
+            for content_item in content_items
+            if (
+                (not content_item.search_window)
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/CR_validators/CR102_validate_execution_mode_search_window.py:32]: a or a and b

is the same as just a
something looks weird with the logic here, double check this

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**